onnxruntime/core/providers/dnnl/dnnl_execution_provider.cc

```diff
@@ -5,42 +5,39 @@
 #pragma warning(disable : 4996)
 #endif
 
-#include "core/framework/allocator.h"
-#include "core/framework/compute_capability.h"
-#include "core/framework/kernel_registry.h"
-#include "core/providers/dnnl/subgraph/dnnl_func_kernel.h"
+#include "core/providers/shared_library/provider_author.h"
+#include <unordered_set>
+#include "subgraph/dnnl_func_kernel.h"
 #include "dnnl_execution_provider.h"
 #include "dnnl_fwd.h"
 
-namespace {
+namespace onnxruntime {
+
 struct KernelRegistryAndStatus {
-  std::shared_ptr<onnxruntime::KernelRegistry> kernel_registry = std::make_shared<onnxruntime::KernelRegistry>();
+  std::shared_ptr<onnxruntime::Provider_KernelRegistry> kernel_registry{Provider_KernelRegistry::Create()};
```

Comment:
nit: underscore separated class names don't conform to the conventions.

Reply:
Yeah, I wanted it to be Provider_<the original class name> as a kind of namespace. I want to change it to a namespace.